components/raftstore/src/store/worker/region.rs

```diff
@@ -895,11 +935,19 @@ where
                 start_key,
                 end_key,
             } => {
-                fail_point!("on_region_worker_destroy", true, |_| {});
-                // try to delay the range deletion because
-                // there might be a coprocessor request related to this range
-                self.insert_pending_delete_range(region_id, start_key, end_key);
-                self.clean_stale_ranges();
+                let region_cleaner = self.region_cleaner.clone();
+                self.region_cleanup_pool
+                    .spawn(async move {
+                        fail_point!("on_region_worker_destroy", region_id == 1000, |_| {});
+                        let mut region_cleaner = region_cleaner.lock().unwrap();
```

Comment:
Currently use a mutex to control the concurrency of `region_cleaner` to 1 is ok, I'm also curious about what if we want to use multiple threads to do multiple regions clean in the future?

Reply:
The mutex over `region_cleaner` is mostly meant to protect the `pending_delete_range` field; other fields like the kv engine and the snapshot manager should be thread safe. If we consider doing a multi-threaded region cleanups, I think we can remove the mutex over `region_cleaner` and put it on the `pending_delete_range` field. But that will probably require more changes to coordinate the cleanups between threads. For example, we probably don't want to delete the same range twice.